modules/lvgl/lvgl.c

```diff
@@ -244,6 +248,27 @@ struct k_work_q *lvgl_get_workqueue(void)
 }
 #endif /* CONFIG_LV_Z_RUN_LVGL_ON_WORKQUEUE */
 
+#ifdef CONFIG_LV_Z_LVGL_MUTEX
+
+static K_MUTEX_DEFINE(lvgl_mutex);
+
+void lvgl_lock(void)
+{
+	(void)k_mutex_lock(&lvgl_mutex, K_FOREVER);
```

Comment:
There is a special case if `LV_Z_USE_OSAL` is active: https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/lvgl/Kconfig#L195C8-L195C21

LVGL also maintains a mutex in its global state. I guess if that option is selected we can save a bit of rom and reuse that one:
https://github.com/lvgl/lvgl/blob/master/src/misc/lv_timer.c#L81